andela/wg-beef-sauce
# [Ft #159356728]Add support for periodization

#### What does this PR do?
Add support for macrocycle, mesocycle and microcycle. Macro is an annual plan, meso is phase of training with a duration of between 2-6 weeks and micro is instead only a week.

#### Description of Task to be completed?
 - A user should be able to create a schedule for one of the periodisation cycles, ie. Macrocycle, mesocycle or microcycle. This is achieved by specifying the desired period when adding a new schedule
 - The duration of workouts added to a cycle depends on the cycle type. Eg. a mesocycle will not allow  a user to add workouts with a total duration exceeding 6 weeks. 

#### How should this be manually tested?
 - Once logged in to the app, a user may create a workout. 
 - The user may then create a new schedule specifying desired cycle type eg. mesocycle.
 - Once the schedule is created, the user should add the created workout, setting a duration higher than 6 weeks
 - The user should be alerted of invalid duration as it should not last more than 6 weeks.
#### Any background context you want to provide?
This feature is needed for sports periodization ( powerlifting, weightlifting, strongman, etc ) or at least to create a fitness programs drafted on more than only one week.
Cycles can be divided into three/four segments, especially for Macrocycles, and so it is recommended they do not exceed 30 weeks.

#### What are the relevant pivotal tracker stories?
[#159356728: Add support for macrocycle, mesocycle and microcycle](https://www.pivotaltracker.com/story/show/159356728)

#### Screenshots:
```Add schedule form showing period selection```
<img width="1258" alt="screen shot 2018-08-16 at 13 11 48" src="https://user-images.githubusercontent.com/37480490/44203016-5e878800-a156-11e8-87c3-c9b898aa1226.png">

```Individual schedule view showing cycle type(ie. Macrocycle) ```
<img width="1111" alt="screen shot 2018-08-16 at 13 13 31" src="https://user-images.githubusercontent.com/37480490/44203032-6a734a00-a156-11e8-800c-f29d4df78f8c.png">

```Example of error message when adding workouts to schedule with total cycle duration exceeded(1 year for macrocycle)```
<img width="1223" alt="screen shot 2018-08-16 at 13 13 21" src="https://user-images.githubusercontent.com/37480490/44203043-74954880-a156-11e8-9329-c29a8e248ec9.png">
